sch_yang_condition.c

```diff
@@ -0,0 +1,923 @@
+    {
+        node = (xpath_node *) iter->data;
+        sch_yang_process_xnode (instance, root, path, step_path, node, &result,
+                                depth + 1, flags);
+        if (result.result)
+        {
+            ptr = strrchr (result.value, '/');
+            if (ptr)
+            {
+                presult->value = g_strdup (ptr + 1);
+                presult->result = result.result;
+            }
+        }
+        g_free (result.value);
+    }
+}
+
+static void
+sch_yang_process_if_feature (sch_instance *instance, GNode *root, char *path,
+                             char *feature, yang_result *presult, int *flags)
+{
+    sch_node *s_node;
+    xmlChar *features = NULL;
+
+    presult->result = false;
+    s_node = sch_lookup (instance, path);
+    if (!s_node)
+        return;
+
+    while (s_node && !features)
+    {
+        features = xmlGetProp ((xmlNode *) s_node, BAD_CAST "features");
+        s_node = sch_node_parent (s_node);
+    }
+
+    if (features && features[0] != '\0')
+    {
+        if (strstr ((char *) features, feature))
+            presult->result = true;
+    }
+    xmlFree (features);
+}
+
+static void
+sch_yang_function_if_feature (sch_instance *instance, GNode *root, char *path,
+                              char *step_path, xpath_node *xnode, yang_result *presult,
+                              int depth, int *flags)
+{
+    GList *iter;
+    xpath_node *node;
+    yang_result result = { };
+
+    presult->result = false;
+    if (g_list_length (xnode->arg_list) != 1)
+        return;
+
+    iter = g_list_first (xnode->arg_list);
+    if (iter)
+    {
+        node = (xpath_node *) iter->data;
+        *flags |= PROC_F_IF_FEATURE;
+        sch_yang_process_xnode (instance, root, path, step_path, node, &result,
+                                depth + 1, flags);
+        *flags &= ~PROC_F_IF_FEATURE;
+        presult->result = result.result;
+        g_free (result.value);
+    }
+}
+
+static int
+sch_yang_function_count_tree_matches (GNode *tree, char *path)
+{
+    GNode *child;
+    int count = 0;
+
+    for (child = tree->children; child; child = child->next)
+        count++;
+
+    return count;
+}
+
+static void
+sch_yang_function_count (sch_instance *instance, GNode *root, char *path, char *step_path,
+                         xpath_node *xnode, yang_result *presult, int depth, int *flags)
+{
+    GList *iter;
+    xpath_node *lnode;
+    GNode *tree;
+    yang_result result = { };
+    int match_count = 0;
+
+    presult->result = false;
+    iter = g_list_first (xnode->arg_list);
+    if (!iter)
+        return;
+
+    lnode = (xpath_node *) iter->data;
+    sch_yang_process_xnode (instance, root, path, step_path, lnode, &result,
+                            depth + 1, flags);
+    if (result.value)
+    {
+        tree = apteryx_get_tree (result.value);
+        if (tree)
+        {
+            match_count += sch_yang_function_count_tree_matches (tree, result.value);
```

Comment:
this and the iter code above implies that there should be some iteration going on here, but there isn't?

Reply:
Removed the function and used g_node_n_children to directly count the tree children.